kornia/color/gray.py

```diff
@@ -27,11 +29,23 @@ def rgb_to_grayscale(image: torch.Tensor) -> torch.Tensor:
         raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
                          .format(image.shape))
 
+    if not isinstance(rgb_weights, torch.Tensor):
+        raise TypeError("rgb_weights is not a torch.Tensor. Got {}".format(
+            type(rgb_weights)))
+
+    if len(rgb_weights.shape) != 1 or rgb_weights.shape[0] != 3:
+        raise ValueError("rgb_weights must have a shape of (3). Got {}"
+                         .format(rgb_weights.shape))
+
+    if not torch.isclose(torch.sum(rgb_weights), torch.tensor(1.0)):
+        raise ValueError("The sum of rgb_weights must be 1. Got {}"
+                         .format(torch.sum(rgb_weights)))
+
     r: torch.Tensor = image[..., 0:1, :, :]
     g: torch.Tensor = image[..., 1:2, :, :]
     b: torch.Tensor = image[..., 2:3, :, :]
 
```

Comment:
```suggestion
w_tmp: torch.Tensor = rgb_weights.to(image.device, image.dtype)
```
weights must be in the shame device/dtype as the input